src/components/forms/SelectField/AsyncSelectField.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import {useAsync} from 'react-use';
+
+import {SelectField} from './SelectField';
+
+const withOptions = WrappedComponent => {
+  return ({optionsRetriever, ...props}) => {
+    const {loading, error, value: options} = useAsync(optionsRetriever);
+
+    if (loading) {
+      return (
+        <WrappedComponent
+          options={[{label: 'Loading...', value: null}]} // Display loading message
```

Comment:
we don't need to handle this ourselves, instead we can pass `isLoading={loading}` to the underlying react-select, see: https://github.com/open-formulieren/formio-builder/blob/main/src/components/builder/prefill/plugin.tsx#L54

Similarly, all user-facing texts should be properly localized using react-intl, currently we support NL/EN but more languages will follow. You can use react-select's [`loadingMessage`](https://react-select.com/props) prop for this.